### PR TITLE
shader: Implement miss attribute layer

### PR DESCRIPTION
**Pull request by vonchenplus**
Fix Pokemon Scarlet/Violet Black screen.

image

### CODE REVIEW

File: src/shader_recompiler/backend/spirv/spirv_emit_context.cpp [hunk 1]
```diff
@@ -1355,6 +1355,10 @@ void EmitContext::DefineInputs(const IR::Program& program) {
     if (loads[IR::Attribute::PrimitiveId]) {
         primitive_id = DefineInput(*this, U32[1], false, spv::BuiltIn::PrimitiveId);
     }
+    if (loads[IR::Attribute::Layer]) {
+        AddCapability(spv::Capability::Geometry);
+        layer = DefineInput(*this, U32[1], false, spv::BuiltIn::Layer);
```
Comment (liamwhite) [inline src/shader_recompiler/backend/spirv/spirv_emit_context.cpp:1360]: `layer` is already used for the output layer; theoretically if a stage both reads from and writes to the layer (e.g. with ShaderViewportIndexLayerEXT), wouldn't this cause problems? I would suggest defining a new variable as `input_layer` here.

File: src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp [hunk 2]
```diff
@@ -315,6 +315,8 @@ Id EmitGetAttribute(EmitContext& ctx, IR::Attribute attr, Id vertex) {
     switch (attr) {
     case IR::Attribute::PrimitiveId:
         return ctx.OpBitcast(ctx.F32[1], ctx.OpLoad(ctx.U32[1], ctx.primitive_id));
+    case IR::Attribute::Layer:
+        return ctx.OpBitcast(ctx.F32[1], ctx.OpLoad(ctx.U32[1], ctx.layer));
```
Comment (liamwhite) [inline src/shader_recompiler/backend/spirv/emit_spirv_context_get_set.cpp:319]: See comment about `input_layer`

**Review by liamwhite — CHANGES_REQUESTED** [review]

File: src/shader_recompiler/backend/spirv/spirv_emit_context.cpp (hunk 1 above)
Comment (vonchenplus) [reply src/shader_recompiler/backend/spirv/spirv_emit_context.cpp:1360]: According to the opengl documentation, there will be no simultaneous reading or writing

**Review by liamwhite — APPROVED** [review]

**Comment by yvancadi** [comment]
can i get the link of thge game

**Comment by v1993** [comment]
> can i get the link of thge game

Sure! https://www.nintendo.com/store/products/pokemon-scarlet-switch/ and https://www.nintendo.com/store/products/pokemon-violet-switch/

**Comment by yvancadi** [comment]
the link of the game on yuzu

**Pull request merged**